File: 1005-univalued-binary-tree/univalued-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isUnivalTree(TreeNode* root) {
        int val = root->val;
        return roots(root,val);}
    bool roots(TreeNode* &root, int &val){
        if(!root){
            return true;
        }
        return roots(root->left,val) && roots(root->right,val) && (root->val == val);
    }
};

File: 1005-univalued-binary-tree/univalued-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isUnivalTree(self, root: Optional[TreeNode]) -> bool:
        res = []
        def roots(node):
            if node:
                res.append(node.val)
                roots(node.left)
                roots(node.right)
        roots(root)
        k = set(res)
        if(len(k)==1):
            return True
        else:
            return False